includes/class-wc-calypso-bridge-introductory-offers.php

```diff
@@ -0,0 +1,84 @@
+<?php
+/**
+ * Contains the logic for querying plans with introductory offers.
+ *
+ * @package WC_Calypso_Bridge/Classes
+ * @since   1.0.6
+ * @version 2.0.8
+ */
+
+use Automattic\Jetpack\Connection\Client;
+use Automattic\Jetpack\Connection\Manager;
+
+/**
+ * WC_Calypso_Bridge_Introductory_offers
+ */
+class WC_Calypso_Bridge_Introductory_offers {
+
+	CONST TRANSIENT_PREFIX = 'wc-calypso-bridge-introductory-plans-';
+
+	/**
+	 * return introductory offers for the current blog.
+	 *
+	 * @return array|mixed
+	 */
+	public static function get_introductory_offers_for_current_blog() {
+		return static::get_introductory_offers( Jetpack_Options::get_option( 'id' ) );
+	}
+
+	/**
+	 * Format offer data for JS.
+	 *
+	 * @param $offer
+	 *
+	 * @return array
+	 */
+	public static function extract_offer_data_for_js( $offer ) {
+		return array(
+			'rawPrice' => $offer['introductory_offer_raw_price'],
+			'formattedPrice' => $offer['introductory_offer_formatted_price'],
+			'intervalUnit' => $offer['introductory_offer_interval_unit'],
+			'intervalCount' => $offer['introductory_offer_interval_count'],
+			'formattedIntervalUnit' => $offer['introductory_offer_interval_count'] === 1 ? $offer['introductory_offer_interval_unit'] : $offer['introductory_offer_interval_unit'].'s',
+		);
+	}
+
+	/**
+	 * Return introductory offers by blog I.D
+	 *
+	 * @param $blog_id
+	 *
+	 * @return array|mixed
+	 */
+	public static function get_introductory_offers( $blog_id ) {
+		if ( ! ( new Manager() )->is_user_connected() ) {
```

Comment:
I think in addition to being connected, we need to check if the current connection is owned by the user since:

1. We're using the blog connection's token to run the request.
2. There can be multiple Jetpack users as admin.

https://github.com/Automattic/jetpack/blob/30a0b584e44affc446ca73678fffc4ed3ba531a6/projects/packages/connection/src/class-manager.php#L727

Reply:
Updated in https://github.com/Automattic/wc-calypso-bridge/pull/1397/commits/d312cb3c21d5339fe8af4d89d4bb864e5085d71c